version.go

```
@@ -52,13 +54,11 @@ func Version() string {
 }
 
 // normalizeVerString returns the passed string stripped of all characters
-// which are not valid according to the semantic versioning guidelines for
-// pre-release version and build metadata strings.  In particular they MUST
-// only contain characters in semanticAlphabet.
-func normalizeVerString(str string) string {
+// which are not valid according to the given alphabet.
+func normalizeVerString(str, alphabet string) string {
 	var result bytes.Buffer
 	for _, r := range str {
-		if strings.ContainsRune(semanticAlphabet, r) {
+		if strings.ContainsRune(alphabet, r) {
```

isn't it better to just return an error if this is not true?

---

Perhaps, yes. But that'd require us to thread through the error in a few places and bug the user with an error for something that's just a nice to have (in case of the initiator). Or do you mean for the semantic version as well?

---

Was thinking just in general. If an invalid user agent or semantic is set, we will fail to start. But perhaps I'm missing some case this would be disruptive for.

---

Okay, I added a check for the version and agent name on startup and also when overwriting the agent name.
I think we really don't care enough about the initiator being silently truncated for an error to be threaded through so I didn't change that behavior.

---

I take it back, better to do what you had initially :)

---

Okay, I reverted the change with the `init()` function but left the check in for when the agent name is overwritten externally.